lib/elixir/lib/gen_event.ex

```diff
@@ -1,11 +1,40 @@
 defmodule GenEvent do
-  # TODO: Deprecate by 1.6 (hard deprecation)
+  # TODO: Remove by 2.0
+
+  # Functions from this module are deprecated in elixir_dispatch.
 
   @moduledoc """
   WARNING: this module is deprecated.
 
-  If you need the same functionality, you can use the `:gen_event` Erlang
```

Comment:
Let's add a presection here that says:

```
If you are interested in implementing an event manager, please read the "Alternatives" section below.
If you have to implement an event handler to integrate with an existing system, such as Elixir's Logger,
please use `:gen_event` instead.
```

This means we need to deprecate `use GenEvent` because the second case will never be picked up otherwise (they don't call any of the `GenEvent` functions).